lib/bundler/definition.rb

```diff
@@ -974,10 +974,11 @@ def compute_requires
 
     def additional_base_requirements_for_resolve
       return [] unless @locked_gems && Bundler.feature_flag.only_update_to_newer_versions?
-      dependencies_by_name = dependencies.group_by(&:name)
+      dependencies_by_name = dependencies.each_with_object({}) {|dep, memo| memo[dep.name] = dep }
+      all_locked_deps = @locked_gems.dependencies
```

Comment:
I'd rather lose this assignment and use the `@locked_gems.dependencies` directly in the loop